homeassistant/util/yaml.py

```diff
@@ -162,14 +160,14 @@ def _secret_yaml(loader: SafeLineLoader,
                                   " but 'logger: %s' found", logger)
                 del secrets['logger']
         else:
-            loader._SECRET_CACHE[secret_path] = None
-    secrets = loader._SECRET_CACHE[secret_path]
+            _SECRET_CACHE[secret_path] = {}
+    secrets = _SECRET_CACHE[secret_path]
 
     # Retrieve secret, first from secrets.yaml, then from keyring
     if secrets is not None and node.value in secrets:
         _LOGGER.debug('Secret %s retrieved from secrets.yaml.', node.value)
         return secrets[node.value]
-    for sname, sdict in loader._SECRET_CACHE.items():
+    for sname, sdict in _SECRET_CACHE.items():
```

Comment:
I think that this causes unexpected behavior. It can now mean that a secret referenced in the file `config/beer/hello/something.yaml` can be defined in `config/wine/secrets.yaml`. I would expect it to search directories in the order of current, parent, parent of parent etc.